contracts/Comptroller.sol

```diff
@@ -1048,6 +1077,35 @@ contract Comptroller is ComptrollerV5Storage, ComptrollerInterface, ComptrollerE
     function _become(Unitroller unitroller) public {
         require(msg.sender == unitroller.admin(), "only unitroller admin can change brains");
         require(unitroller._acceptImplementation() == 0, "change not authorized");
+
+        // TODO: Remove this post upgrade
+        Comptroller(address(unitroller))._upgradeSlitCompRewards();
+    }
+
+    function _upgradeSlitCompRewards() public {
+        require(msg.sender == comptrollerImplementation, "only brains can become itself");
+
+        // compSpeeds -> compBorrowSpeeds & compSupplySpeeds
+        for (uint i = 0; i < allMarkets.length; i ++) {
+            compBorrowSpeeds[address(allMarkets[i])] = compSupplySpeeds[address(allMarkets[i])] = compSpeeds[address(allMarkets[i])];
+            delete compSpeeds[address(allMarkets[i])];
+
+            /*
+             * Ensure supply and borrow state indices are all set. If not set, update to default value
+             */
+            CompMarketState storage supplyState = compSupplyState[address(allMarkets[i])];
+            CompMarketState storage borrowState = compBorrowState[address(allMarkets[i])];
+
+            if (supplyState.index == 0) {
+                // Initialize supply state index with default value
+                supplyState.index = compInitialIndex;
+            }
+
+            if (borrowState.index == 0) {
+                // Initialize borrow state index with default value
+                borrowState.index = compInitialIndex;
+            }
```

Comment:
We don't really need to do it because those block numbers are updated regardless of if the market is getting COMP rewards. But I added such code anyway as there's no harm in doing so, just to ensure everything is up-to-date.